src/UrlHelper.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Rokka\Client;
+
+class UrlHelper
+{
+    /**
+     * Allows you to add stack options to a Rokka URL.
+     *
+     * Useful eg. if you just want to add "options-dpr-2" to an existing URL
+     *
+     * @param string $url     The rokka image render URL
+     * @param string $options The options you want to add as string
+     *
+     * @return string
+     */
+    public static function addOptionsToUrl($url, $options)
+    {
+        $components = parse_url($url);
+        if (preg_match('#^/(?<stack>[^/]+)/(?<rest>[0-9a-f]{40}.*)$#', $components['path'], $matches)) {
+            // nothing to do here
+        } elseif (preg_match('#^/(?<stack>[^/]+)/(?<options>[^/]+)/(?<rest>[0-9a-f]{40}.*)$#', $components['path'], $matches)) {
+            $urlOptions = self::decomposeOptions($matches['options']);
+            $inputOptions = self::decomposeOptions($options);
+            $combinedOptions = array_replace_recursive($urlOptions, $inputOptions);
+            $newOptions = [];
+            foreach ($combinedOptions as $key => $values) {
+                $newOption = "$key";
+                foreach ($values as $k => $v) {
+                    $newOption .= "-$k-$v";
+                }
+                $newOptions[] = $newOption;
+            }
+            $options = implode('--', $newOptions);
+        }
+        $newUrl = $components['scheme'].'://'.$components['host'].'/'.$matches['stack'].'/'.$options.'/'.$matches['rest'];
```

Comment:
Yep, that's fine. We use the rokka php client anyway with all it's dependencies. I was not really clear above, I assume. I just need a "function" in those templates without having to need to instantiate objects. But in the end `\Rokka\Client\UrlHelper::addOptionsToUri(new \GuzzleHttp\Psr7\Uri($url), $options)` would have worked there as well, just way more to write